src/main/java/org/jenkinsci/plugins/googleplayandroidpublisher/ReleaseTrackAssignmentBuilder.java

```diff
@@ -19,96 +21,165 @@
 import java.io.PrintStream;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 import java.util.TreeSet;
-import java.util.zip.ZipException;
+import java.util.stream.Collectors;
 
 import static hudson.Util.fixEmptyAndTrim;
 import static hudson.Util.tryParseNumber;
 import static org.jenkinsci.plugins.googleplayandroidpublisher.Constants.PERCENTAGE_FORMATTER;
 import static org.jenkinsci.plugins.googleplayandroidpublisher.ReleaseTrack.fromConfigValue;
 import static org.jenkinsci.plugins.googleplayandroidpublisher.Util.getPublisherErrorMessage;
-import static org.jenkinsci.plugins.googleplayandroidpublisher.Util.getVersionCode;
 
 public class ReleaseTrackAssignmentBuilder extends GooglePlayBuilder {
 
-    @DataBoundSetter
-    private Boolean fromVersionCode;
+    @VisibleForTesting Boolean fromVersionCode;
+    @VisibleForTesting String applicationId;
+    @VisibleForTesting String versionCodes;
+    private String filesPattern;
+    @VisibleForTesting String trackName;
+    @VisibleForTesting Double rolloutPercent;
 
-    @DataBoundSetter
-    protected String applicationId;
+    @Deprecated private transient String apkFilesPattern;
+    @Deprecated private transient String rolloutPercentage;
```

Comment:
I'll open an issue for this, and if I get time submit a PR as well. But with this deprecation, and replacement to `Double`, it's now no longer possible to set a build parameter as the rollout percentage.

Previously, we were using a build parameter `ROLLOUT` that could be specified at build time to choose various rollout percentage options.  The build step would then be defined by specifying the percentage as `${ROLLOUT}`, and that would be expanded at build time.

Now, since `${ROLLOUT}` is not a number, it defaults to `100.0%` regardless of what was selected at build time. That causes a build that was intended to be rolled out at a small percentage, to be published straight to 100% unexpectedly.